modules/aleph/includes/aleph.debt.inc

```diff
@@ -5,8 +5,49 @@
  * Handles debt with Aleph.
  */
 
+use Drupal\aleph\Aleph\Handler\AlephPatronHandler;
+
+/**
+ * Return patron debts.
+ *
+ * @return array
+ *    Array of DingProviderDebt objects containing debts for the patron.
+ */
 function aleph_debt_list() {
-  return array();
+  $result = array();
+  $response = array();
+
+  $patron = aleph_get_patron_by_global_account();
+  $patron_handler = new AlephPatronHandler(aleph_client());
+  $patron_handler->setPatron($patron);
+
+  try {
+    $response = $patron_handler->getDebts();
+  }
+  catch (Exception $e) {
+    watchdog_exception('aleph', $e);
+  }
+
+  $debts = $response->xpath('charges/institution/cash');
+
+  $id = 0;
+  foreach ($debts as $debt) {
+    $id++;
+    $amount = str_replace(array('(', ')'), '', (string) $debt->xpath('z31/z31-sum')[0]);
+    $data = array(
+      'date' => (string) $debt->xpath('z31/z31-date')[0],
```

Comment:
What's the format of the date string? Does it match the date display format of the site?